File: src/main/java/com/tms/csp/transforms/OneArgAnd.java
package com.tms.csp.transforms;

import com.tms.csp.ast.Exp;

/**
 * Replaces each And(a) with a
 */
public class OneArgAnd extends BaseTransformer {

    @Override
    protected Exp local(Exp in) {
        return in.getArg();
    }

    @Override
    protected boolean executeLocal(Exp in) {
        return in.isAnd() && in.argCount() == 1;
    }

}

File: src/main/java/com/tms/csp/transforms/OneArgAnd.kt
package com.tms.csp.transforms

import com.tms.csp.ast.Exp

/**
 * Replaces each And(a) with a
 */
class OneArgAnd : BaseTransformer() {

    override fun local(inExp: Exp): Exp? {
        return inExp.arg
    }

    override fun executeLocal(inExp: Exp): Boolean {
        return inExp.isAnd && inExp.argCount() == 1
    }

}
